mlflow/transformers/model_io.py

```diff
@@ -131,10 +131,33 @@ def _load_model(model_name_or_path, flavor_conf, accelerate_conf, device, revisi
       3. Load the model without the device
     """
     import transformers
+    from transformers import AutoConfig
 
-    cls = getattr(transformers, flavor_conf[FlavorKey.MODEL_TYPE])
+    if hasattr(transformers, flavor_conf[FlavorKey.MODEL_TYPE]):
+        cls = getattr(transformers, flavor_conf[FlavorKey.MODEL_TYPE])
+        trust_remote = False
+    else:
+        config = AutoConfig.from_pretrained(
+            model_name_or_path, revision=revision, trust_remote_code=True
+        )
+        class_name = config.architectures[0]
+
+        if hasattr(transformers, class_name):
+            cls = getattr(transformers, class_name)
+            trust_remote = False
+        else:
+            auto_classes = [
+                auto_class for auto_class, module in config.auto_map.items() if class_name in module
+            ]
+            if len(auto_classes) == 0:
+                raise MlflowException(f"Couldn't find a loader class for {class_name}")
+            auto_class = auto_classes[0]
+            cls = getattr(transformers, auto_class)
```

Comment:
The autoclass implementation in transformers will use inspect to determine if the classdef is loaded within context? If so, this is a brilliant fix. 
Can we add a very lightweight custom model implementation that refers to a tiny test-only transformers model implementation for the unit test?

Reply:
We might want to extract the "fetch of the transformers class" to a utility function to help improve readability and to have that function handle the assignment of "trust_remote_code" as a compound return type. The only reason I bring it up is that while this function is now easily readable for those of us who have now touched this flavor's implementation, it might be a bit complex for a newcomer to reason about what is going on here :)

Reply:
yup pretty much, transformers uses the `auto_map` field in the config file to determine which module they should use to load the class. they handle all the necessary stuff (e.g. loading custom module code from HF hub/cache, + all that dynamic module creation stuff) in order to get the model to load.

Reply:
for the test, would it suffice if i added a small model that uses custom code from the hub? e.g. test saving and loading this model: https://huggingface.co/ccdv/lsg-bert-base-uncased-4096/tree/main